genie-web/src/test/groovy/com/netflix/genie/web/configs/GenieAgentFilterAutoConfigurationSpec.groovy

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.web.configs
+
+import com.netflix.genie.web.properties.AgentFilterProperties
+import spock.lang.Specification
+
+class GenieAgentFilterAutoConfigurationSpec extends Specification {
```

Comment:
Look into Spring's `ApplicationContextRunner`. I've used it in a few places to test these auto configuration classes. it lets you verify your conditionals work as expected etc. you can find some examples in the code base or I can show you.